graphql/resolvers.go

```diff
@@ -1137,18 +1137,39 @@ func (r *patchResolver) ID(ctx context.Context, obj *restModel.APIPatch) (string
 	return *obj.Id, nil
 }
 
-func (r *patchResolver) PatchTriggerAliases(ctx context.Context, obj *restModel.APIPatch) ([]*restModel.APIPatchTriggerDefinition, error) {
-	project, err := r.sc.FindProjectById(*obj.ProjectId, true)
-	if err != nil || project == nil {
+func (r *patchResolver) PatchTriggerAliases(ctx context.Context, obj *restModel.APIPatch) ([]*restModel.APIPatchTriggerAlias, error) {
+	projectRef, err := r.sc.FindProjectById(*obj.ProjectId, true)
+	if err != nil || projectRef == nil {
 		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("Could not find project: %s : %s", *obj.ProjectId, err))
 	}
 
-	aliases := []*restModel.APIPatchTriggerDefinition{}
-	for _, alias := range project.PatchTriggerAliases {
-		aliases = append(aliases, &restModel.APIPatchTriggerDefinition{
-			Alias:        utility.ToStringPtr(alias.Alias),
-			ChildProject: utility.ToStringPtr(alias.ChildProject)})
+	project := &model.Project{}
+	opts := model.GetProjectOpts{}
+	if _, err = model.LoadProjectInto(ctx, []byte(*obj.PatchedConfig), opts, *obj.ProjectId, project); err != nil {
```

Comment:
Parsing yaml can be really expensive; if `len(projectRef.PatchTriggerAliases) == 0` we should return early so we don't need to go through this.

Reply:
If this is only used for the Configure Page (I think it is, Sophie would know) then I imagine it's not a concern since it would only increase the time for that one page, right? 
Otherwise if we also need this for the regular patch page, if the patch has been finalized we can read the project from the parser project collection, which is less expensive. I think this isn't the scenario though.

Reply:
Yes, it's used only for the configure page. This data will not be needed anywhere else because once the child patch is scheduled, its variants/tasks are available via the Patch resolver's `childPatches` field.

I agree that this could be expensive, but I'm not very in tune with the usage patterns of the configure page. It seems to me like this may be something that a user looks at once, schedules the patch, and doesn't look at again. In that case the caching may not be particularly helpful, but it's tough to speculate. Would really appreciate any other insights y'all have!

Reply:
Sorry our comments kind of raced in a weird way 😂  The configure page isn't used often, and most projects won't define patch trigger aliases, so I personally think a mild slowdown on this page is fine.

Reply:
Cool SGTM. Thanks for the explanation